### PR TITLE
Terminal Access

### PR DESCRIPTION
**Pull request by gamaanderson**
Add a plugin to give the user direct access to artview, so he can change the data directly in python. It is a powerful tool for some one that knows what he is doing, but it can be dangerous as well if not used carefully. There may be some ways of making it safer, but it involves much more code for emulating a terminal, however the idea itself of giving a terminal to someone is risk.

### CODE REVIEW

**Comment by nguy** [comment]
This is part of the reason I included the ```Show Script``` in the upcoming gatefilter (still working on this on the pyart side). This way we could produce the artview calls that could then be used in a terminal window. It definitely has it's risks, but unless the user save the newly produced data I am inclined to say it's benign.

That being said, I have no problem exploring this further.

**Comment by nguy** [comment]
Is this something you'd like to merge or wait?

**Comment by gamaanderson** [comment]
This is more than given a script, it give total control over artview at execution time, one easily open a new display, a new radar or modify the existing ones and seeing the result direct on the running artview.

I think this can be merged.

File: artview/plugins/console.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        print artview
+        banner = ("HELLO: this is an iteractive python console so you can\n" +
+                  "acess ARTview from the incide and manipulate the data\n" +
+                  "directly. You have acess to two variables:\n" +
+                  "    'components': List of all running components\n" +
+                  "    'pyart': Python-ARM Radar Toolkit.\n" +
+                  "    'artvew': ARM Radar Toolkit Viewer\n" +
+                  "To leave and go back to grafical ARTview press ctrl+D\n" +
+                  "in unix and OXS or ctrl+Z in Windows.")
+        thread = threading.Thread(
```
Comment (nguy) [inline artview/plugins/console.py]: Can we change banner to the following?

```
        banner = ("HELLO: this is an iteractive python console so you can\n" +
                  "access ARTview functions while running the GUI and manipulate the data directly.\n\n" 
                  " You have acess to two variables:\n" +
                  "    'components': A list of all running components\n" +
                  "    'pyart': Python-ARM Radar Toolkit.\n" +
                  "    'artview': ARM Radar Toolkit Viewer\n\n" +
                  "To leave and go back to graphical ARTview press ctrl+D\n" +
                  "in Unix/OXS or ctrl+Z in Windows.")

File: artview/plugins/console.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    'readfunc': None,
+                    'local': {'components': core.componentsList, 'pyart': pyart,
+                              'artview': artview}}
+            )
+        thread.start()
+        thread.join()
+
+
+
+_plugins = [AccessTerminal]
```
Comment (nguy) [inline artview/plugins/console.py]: Can we also add an exit message to print to terminal:
```print("Ending interactive session, adios!"

Or something equally cheeky. :)

File: artview/plugins/console.py (hunk 1 above)
Comment (gamaanderson) [inline artview/plugins/console.py]: yes, of course, just correct for me "two" to "three variables"

File: artview/plugins/console.py (hunk 2 above)
Comment (gamaanderson) [inline artview/plugins/console.py]: as you wish.

File: artview/plugins/console.py (hunk 1 above)
Comment (nguy) [inline artview/plugins/console.py]: Done and merging.

**Pull request merged**